common/src/test/java/org/apache/drill/test/SubDirTestWatcher.java

```diff
@@ -83,10 +84,10 @@
   private List<Path> subDirs;
 
   protected SubDirTestWatcher(File baseDir, boolean createAtBeginning, boolean deleteAtEnd, List<Path> subDirs) {
-    this.baseDir = Preconditions.checkNotNull(baseDir);
+    this.baseDir = Objects.requireNonNull(baseDir);
     this.createAtBeginning = createAtBeginning;
     this.deleteAtEnd = deleteAtEnd;
-    this.subDirs = Preconditions.checkNotNull(subDirs);
+    this.subDirs = Objects.requireNonNull(subDirs);
 
     Preconditions.checkArgument(!subDirs.isEmpty(), "The list of subDirs is empty.");
```

Comment:
While in this place `Preconditions.checkNotNull()` and `Objects.requireNonNull()` provide the same functionality, in general guava `Preconditions` supports template error message that `Objects.requireNonNull` does not support. I'd recommend keeping `Preconditions.checkNotNull` for consistency. IMO, it is necessary to remove *deprecated* guava classes, not *all* references to guava where similar functionality is provided by JDK.

Reply:
@vvysotskyi -1: individual preferences do not matter in Apache projects. In the new code that you contribute, you may use `Objects.requireNonNull()`, but there is no technical reason to change existing contributions. To change community preference, the issue must be discussed and voted on dev@drill.

Reply:
@vrozov, I didn't correctly express my point of view: I don't see a reason for using methods from third-party libraries if JDK provides methods with exactly the same functionality. Such usages of third-party libraries should be prevented for the consistency.

Using methods from JDK is also helps to prevent problems with updating library versions, since the API of a third-party library may change, but JDK guarantees backward compatibility.

Reply:
@vvysotskyi I already explained reasons to keep `Preconditions` in my other comments. Here is the summary in random order:
- The change does not remove the dependency on the guava library.
- `Preconditions` are not deprecated by the guava team and provides a stable interface.
- `Preconditions` functionality is much wider compared to JDK classes.
- Using `Preconditions` will be more consistent with other checks like `checkState` or `checkArgument` and also with other Apache projects.
- It avoids unnecessary changes and keeps original contributions.

Reply:
I agree with what @vrozov said; technically, there seems to be no clear benefit of replacing existing `Preconditions`.   Personal preference should only apply to new contributions , not to old contributions (since the original contributor(s) could also have her/his preference).

Reply:
Thank you all for discussing this. I have reverted my changes where I replaced `Preconditions.checkNotNull()` with `Objects.requireNonNull()`